compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -124,6 +124,7 @@ mod erase_regions;
 mod generics;
 mod impls_ty;
 mod instance;
+mod layout_sanity_check;
```

Comment:
<sub><i>(this is most of a comment first written when I started a review on the original state of the PR but didn't submit it right away)</i></sub>

You could deal with the large file by putting the checks in `ty/layout_checks.rs` or `ty/layout/checks.rs` (the latter would mean `mod checks;` in `layout.rs` and would be able to do `use super::*;` to improve everything including private items - I have mixed feelings but it's practical I think).

<sub><i>(thoughts unactionable in this PR follow below)</i></sub>

Honestly we should consider to what crate to move the entire `layout` module, since it's *mostly* just providing queries, and it feels awkward where it is.

I *also* wish it was `rustc_middle::ty::abi` *or* `rustc_middle::abi` (could even go into `mir` but I doubt that makes as much sense) for consistency with `rustc_target::abi` but that's another mess.

Reply:
I wish we used the term 'abi' less, not more, given how ambiguous it is.^^